File: backend/api/src/routes/modules/assignments/starter/get.rs
use crate::{
    response::ApiResponse,
    routes::modules::assignments::starter::common::{STARTER_PACKS, StarterPack},
};
use axum::{Json, extract::State, response::IntoResponse};
use hyper::StatusCode;
use util::state::AppState;

pub async fn list(_: State<AppState>) -> impl IntoResponse {
    // Return as ApiResponse so the frontend’s api utils can consume it.
    let items: Vec<StarterPack> = STARTER_PACKS.iter().cloned().collect();
    (
        StatusCode::OK,
        Json(ApiResponse::success(items, "Starter packs")),
    )
}

File: frontend/src/services/modules/assignments/starter/get.ts
import type { ApiResponse } from "@/types/common";
import type { CreateStarterRequest } from "@/types/modules/assignments/starter";
import { api } from "@/utils/api";

export function createStarter(
  moduleId: number,
  assignmentId: number,
  payload: CreateStarterRequest
): Promise<ApiResponse<unknown>> {
  return api.post<unknown>(
    `/modules/${moduleId}/assignments/${assignmentId}/starter`,
    payload
  );
}
